src/zenml/integrations/pandas/materializers/pandas_materializer.py

```diff
@@ -90,23 +121,49 @@ def load(self, data_type: Type[Any]) -> Union[pd.DataFrame, pd.Series]:
         Returns:
             The pandas dataframe or series.
         """
-        if self.artifact_store.exists(self.parquet_path):
-            if self.pyarrow_exists:
-                with self.artifact_store.open(
-                    self.parquet_path, mode="rb"
-                ) as f:
-                    df = pd.read_parquet(f)
+        try:
+            # First try normal loading
+            if self.artifact_store.exists(self.parquet_path):
+                if self.pyarrow_exists:
+                    with self.artifact_store.open(
+                        self.parquet_path, mode="rb"
+                    ) as f:
+                        df = pd.read_parquet(f)
+                else:
+                    raise ImportError(
+                        "You have an old version of a `PandasMaterializer` "
+                        "data artifact stored in the artifact store "
+                        "as a `.parquet` file, which requires `pyarrow` "
+                        "for reading, You can install `pyarrow` by running "
+                        "'`pip install pyarrow fastparquet`'."
+                    )
             else:
-                raise ImportError(
-                    "You have an old version of a `PandasMaterializer` "
-                    "data artifact stored in the artifact store "
-                    "as a `.parquet` file, which requires `pyarrow` "
-                    "for reading, You can install `pyarrow` by running "
-                    "'`pip install pyarrow fastparquet`'."
+                with self.artifact_store.open(self.csv_path, mode="rb") as f:
+                    df = pd.read_csv(f, index_col=0, parse_dates=True)
+        except Exception as e:
+            # Check for common data type error patterns
+            error_str = str(e).lower()
+
+            is_dtype_error = (
```

Comment:
This is very flaky. The only reason that I can see to even have this code is that the pandas error is too bad, so we have to somehow try to improve it in our library?

Reply:
hmm what would you suggest otherwise

Reply:
Don't know, open up a PR in the pandas repo to improve the error message. Not a ZenML concern IMO

Reply:
I mean i think its still nice to have it

Reply:
It will be:
- above the stack trace. The kind of user that runs into this error and isn't able to figure it out from the pandas exception will for sure not scroll up above and look at some warnings.
- flaky, if the wording of the error message changes at all, it will simply not work anymore.
- untested, which means we will never update this at all
To me this is simply dead code/tech debt without a purpose, but if you think it's helpful feel free to merge. I don't think we need any testing of this.